jobserver/models.py

```diff
@@ -47,7 +47,7 @@ class Job(models.Model):
 
     # The unique identifier created by job-runner to reference this Job.  We
     # trust whatever job-runner sets this to.
-    identifier = models.TextField()
+    identifier = models.TextField(unique=True)
```

Comment:
Why not make it the primary key?

Reply:
And if not for some reason, it should be `indexed` at least?

Reply:
> Why not make it the primary key?

I don't want to couple job-servers implementation of Job storage to that of job-runner.  Experience has shown it's a pain to decouple later on!

> should be indexed at least?

I already is! "unique implies the creation of an index" (from [the docs](https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.Field.unique)).